plugins/custom_caret.lua

```diff
@@ -0,0 +1,46 @@
+-- mod-version:3
+
+--[[
+  Plugin to customize the caret in the editor
+  Thanks to @Guldoman for the initial example
+
+  Features
+    1. Change the Color and Opacity of the caret
+    2. Change the Shape of the caret, available shapes are Line, Block, Underline
+    3. Change the dimensions of the caret
+
+  Customizing the Caret (this can be changed from the .config/lite-xl/init.lua file)
+    1. config.plugins.custom_caret.shape - Change the shape of the caret [string]
+    2. config.plugins.custom_caret.color - Change the color of the caret in rgba [table]
+    3. config.plugins.custom_caret.width - Change the width of the caret [number]
+    4. config.plugins.custom_caret.height - Change the height of the caret [number]
+]]
+
+local core = require "core"
+local style = require "core.style"
+local config = require "core.config"
+local DocView = require "core.docview"
+
+config.plugins.custom_caret = {}
+
+function DocView:draw_caret(x, y)
+  local lh = self:get_line_height()
+  local caret_top_width = math.ceil(self:get_font():get_width("a"))
```

Comment:
as with `config.plugins.custom_caret.height` you can drop `config.plugins.custom_caret.width` and use `caret_top_width` directly which should be maybe renamed to `caret_width`.

Reply:
but I use ```config.plugins.custom_caret.height``` and ```config.plugins.custom_caret.width```, so that they can be changed by the user, if they think the caret is too thick/thin.